### PR TITLE
Fix warning message for indexing into lists, display fully qualified classname in warnings

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

JIRA: https://jira.autodesk.com/browse/DYN-1487, https://jira.autodesk.com/browse/DYN-1044

The warning on the node no longer mentions `Get.ValueAtIndex` and displays the appropriate error message:
image

image

image

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Engine/ProtoCore/RuntimeStatus.cs [hunk 1]
```diff
@@ -408,7 +416,7 @@ private void GetLocationByGraphNode(ref int line, ref int col)
                 var argsJoined = string.Join(", ", arguments.Select(GetTypeName));
                 
                 var fep = funcGroup.FunctionEndPoints[0];
-                var formalParamsJoined = string.Join(", ", fep.FormalParams.Select(x => x.ToShortString()));
+                var formalParamsJoined = string.Join(", ", fep.FormalParams);
```
Comment (mjkkirschner) [inline src/Engine/ProtoCore/RuntimeStatus.cs]: is this going to use full class names when throwing function not found exceptions?

File: src/Libraries/DesignScriptBuiltin/IndexingExceptions.cs [hunk 2]
```diff
@@ -25,4 +25,12 @@ public StringOverIndexingException(string message)
         {
         }
     }
+
+    public class BuiltinNullReferenceException : Exception
```
Comment (mjkkirschner) [inline src/Libraries/DesignScriptBuiltin/IndexingExceptions.cs]: is it worth adding a summary to this?

File: src/Engine/ProtoCore/RuntimeStatus.cs (hunk 1 above)
Comment (aparajit-pratap) [reply src/Engine/ProtoCore/RuntimeStatus.cs]: Yes, this is the intended fix for https://jira.autodesk.com/browse/DYN-1044.

**Comment by mjkkirschner** [comment]
This looks good, just one comment.

**Pull request merged**